core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -264,4 +268,28 @@ package object internal {
       }
     }
   }
+
+  /** Internal API for [[ViewParent]] */
+  sealed private[chisel3] abstract class ViewParentAPI extends RawModule()(ExplicitCompileOptions.Strict) {
```

Comment:
Why does this not extend `PseudoModule` ? I see you mix it in later?

Reply:
Good question, I think it's just due to the order of implementation and partial refactoring. I'm moving all of `ViewParent` into `ViewParentAPI` and making it just a `class`.